server/server.js

```diff
@@ -258,6 +259,66 @@ module.exports = function(port, db, githubAuthoriser) {
         });
     });
 
+    app.post("/api/groups", function(req, res) {
+        var groupInfo = req.body;
+        var creatorID = req.session.user;
+        groups.find({
+            name: groupInfo.name
+        }).limit(1).next().then(function(group) {
+            if (!group) {
+                groups.insertOne({
```

Comment:
I'm wondering if this post method and the put method below can be a bit cleaner. Come over when you have a minute?